src/main/java/com/tproject/dao/impl/MarkDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+package com.tproject.dao.impl;
+
+import com.tproject.dao.MarkDao;
+import com.tproject.entity.Mark;
+import com.tproject.exception.CustomSQLException;
+
+import java.util.*;
+import java.sql.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class MarkDaoImpl implements MarkDao {
+
+    private static MarkDaoImpl instance;
+
+    public static MarkDaoImpl getInstance() {
+        MarkDaoImpl localInstance = instance;
+        if (localInstance == null) {
+            synchronized (MarkDaoImpl.class) {
+                localInstance = instance;
+                if (localInstance == null) {
+                    instance = localInstance = new MarkDaoImpl();
+                }
+            }
+        }
+        return localInstance;
+    }
+
+    private static final Logger LOGGER =
+            Logger.getLogger(MarkDaoImpl.class.getName());
+
+    private Mark composeMark(ResultSet resultSet) throws SQLException {
+        Mark mark = new Mark();
+        mark.setId(resultSet.getInt("id"));
+        mark.setUserProfileId(resultSet.getInt("user_id"));
+        mark.setScore(resultSet.getFloat("score"));
+        mark.setDate(resultSet.getDate("date"));
```

Comment:
is there a getLocalDate methos in resitlSet?

Reply:
no, only getDate().